hermes/integration/app_test.go

```diff
@@ -0,0 +1,652 @@
+			State    string `json:"state"`
+			Version  string `json:"version"`
+		} `json:"channels"`
+	}
+
+	QueryBalances struct {
+		Balances sdk.Coins `json:"balances"`
+	}
+)
+
+func runChain(
+	t *testing.T,
+	ctx context.Context,
+	env envtest.Env,
+	app envtest.App,
+	cfg v1.Config,
+	tmpDir string,
+	ports []uint,
+) (api, rpc, grpc, faucet string) {
+	t.Helper()
+	if len(ports) < 7 {
+		t.Fatalf("invalid number of ports %d", len(ports))
+	}
+
+	var (
+		chainID   = cfg.Genesis["chain_id"].(string)
+		chainPath = filepath.Join(tmpDir, chainID)
+		homePath  = filepath.Join(chainPath, "home")
+		cfgPath   = filepath.Join(chainPath, chain.ConfigFilenames[0])
+	)
+	require.NoError(t, os.MkdirAll(chainPath, os.ModePerm))
+
+	genAddr := func(port uint) string {
+		return fmt.Sprintf(":%d", port)
+	}
+
+	cfg.Validators[0].Home = homePath
+
+	cfg.Faucet.Host = genAddr(ports[0])
+	cfg.Validators[0].App["api"] = yamlmap.Map{"address": genAddr(ports[1])}
+	cfg.Validators[0].App["grpc"] = yamlmap.Map{"address": genAddr(ports[2])}
+	cfg.Validators[0].App["grpc-web"] = yamlmap.Map{"address": genAddr(ports[3])}
+	cfg.Validators[0].Config["p2p"] = yamlmap.Map{"laddr": genAddr(ports[4])}
+	cfg.Validators[0].Config["rpc"] = yamlmap.Map{
+		"laddr":       genAddr(ports[5]),
+		"pprof_laddr": genAddr(ports[6]),
+	}
+
+	file, err := os.Create(cfgPath)
+	require.NoError(t, err)
+	require.NoError(t, yaml.NewEncoder(file).Encode(cfg))
+	require.NoError(t, file.Close())
+
+	app.SetConfigPath(cfgPath)
+	app.SetHomePath(homePath)
+	go func() {
+		env.Must(app.Serve("should serve chain", envtest.ExecCtx(ctx)))
+	}()
+
+	genHTTPAddr := func(port uint) string {
+		return fmt.Sprintf("http://127.0.0.1:%d", port)
+	}
+	return genHTTPAddr(ports[1]), genHTTPAddr(ports[5]), genHTTPAddr(ports[2]), genHTTPAddr(ports[0])
+}
+
+func TestHermes(t *testing.T) {
+	var (
+		require     = require.New(t)
+		env         = envtest.New(t)
+		app         = env.Scaffold("github.com/apps/hermes-app")
+		tmpDir      = t.TempDir()
+		ctx, cancel = context.WithCancel(env.Ctx())
+	)
+	t.Cleanup(func() {
+		cancel()
+		time.Sleep(5 * time.Second)
+		require.NoError(os.RemoveAll(tmpDir))
+	})
+
+	dir, err := os.Getwd()
+	require.NoError(err)
+	pluginPath := filepath.Join(filepath.Dir(filepath.Dir(dir)), "hermes")
+
+	env.Must(env.Exec("install hermes app locally",
+		step.NewSteps(step.New(
+			step.Exec(envtest.IgniteApp, "app", "install", pluginPath),
+			step.Workdir(app.SourcePath()),
+		)),
+	))
+
+	// One local plugin expected
+	assertLocalPlugins(t, app, []pluginsconfig.Plugin{
+		{
+			Path: pluginPath,
+		},
+	})
+	assertGlobalPlugins(t, app, nil)
+
+	// prepare the chain
+	env.Must(env.Exec("create an IBC module",
+		step.NewSteps(step.New(
+			step.Exec(envtest.IgniteApp,
+				"s",
+				"module",
+				"blog",
+				"--ibc",
+				"--require-registration",
+				"--yes",
+			),
+			step.Workdir(app.SourcePath()),
+		)),
+	))
+
+	env.Must(env.Exec("create a post type list in an IBC module",
```

Comment:
@Pantani is it really necessary to create the post types and the custom IBC keeper handlers? I'm wondering if it would be enough to only have a default IBC module in this case that is only testing the relayer with a transfer. If so then we could remove a lot of extra code making the test easier to maintain and follow.